server/src/main/java/org/elasticsearch/transport/TransportActionProxy.java

```diff
@@ -117,27 +122,54 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
+    private static class CancellableProxyRequest<T extends TransportRequest> extends ProxyRequest<T> {
+        CancellableProxyRequest(StreamInput in, Writeable.Reader<T> reader) throws IOException {
+            super(in, reader);
+        }
+
+        @Override
+        public Task createTask(long id, String type, String action, TaskId parentTaskId, Map<String, String> headers) {
+            return new CancellableTask(id, type, action, "", parentTaskId, headers) {
```

Comment:
Rather than requiring every registration to record correctly whether the proxied task is cancellable or not, could we call `wrapped.createTask()` and decide whether to create a `CancellableTask` or just a regular `Task` based on whether the wrapped request's task was cancellable or not? At least we should assert that they match, or else we'll forget to update the argument to the proxy registration in future if we ever add cancellability  to another task.

Reply:
+1, I added an assertion in https://github.com/elastic/elasticsearch/pull/65850/commits/0a52a5b9f695bdee2648cb197b3aa439586a6981